surveys/views.py

```diff
@@ -48,7 +69,11 @@ def get_context_data(self, **kwargs):
         context['surveys_submitted'] = surveys.order_by('form_id', '-time_stop').distinct('form_id')
 
         for survey in context['surveys_submitted']:
-            survey.form_title = FormEntry.objects.get(id=survey.form_id)
+            try:
+                survey_form_entry = SurveyFormEntry.objects.get(formentry_ptr=survey.form_id)
+                survey.form_title = survey_form_entry.name
+            except SurveyFormEntry.DoesNotExist:
+                survey.form_title = "[Deleted Survey]"
```

Comment:
It seems like this logic represents defensive programming in the case that the user has deleted the referenced `SurveyFormEntry`, is that right? I wonder if there's a safer way that we can handle this case. My instinct would be to enforce `ON DELETE CASCADE` so that when a `SurveyFormEntry` gets deleted all of the `Survey`s that reference it will also get deleted, but I understand that we can't enforce that at the ORM level in `django-pldp` because the `Survey.form_id` attribute shouldn't be a foreign key to a separate app.

I wonder if we could [add a custom constraint](https://www.fusionbox.com/blog/detail/custom-database-constraints-in-django/594/) to the Survey table with a migration in the `surveys` app? The constraint I'm imagining would be something like:

```sql
ALTER TABLE pdlp_survey
ADD CONSTRAINT form FOREIGN KEY (form_id)
REFERENCES surveys_surveyform (id) ON DELETE CASCADE
```

I'm not really sure if this would work since I don't know if we can reliably say that the `pldp_survey` table will exist before the `surveys_surveyform` table in the app (I would guess that the order of the apps in `INSTALLED_APPS` would determine this). Still, I think it's worth a shot.

Reply:
I opened issue #84 to figure this out. @jeancochrane, do you want to add your approach there?

Reply:
Done!